e2e-tests/docs/CI.md

```diff
@@ -0,0 +1,135 @@
+- **Steps:**
+  1. **Detection:** OpenShift-CI detects the PR event.
+  2. **Environment Setup:** The test environment is set up using the `openshift-ci-tests.sh` script.
+  3. **Test Execution:**
+     - **Running Tests:** Executes test suites using `yarn` commands specified in `package.json`.
+     - **Retry Logic:** Individual tests are retried up to 2 times as specified in the Playwright configuration.
+  4. **Artifact Collection:**
+     - Collects test artifacts (logs, screenshots, recordings).
+     - Stores artifacts in the designated `ARTIFACT_DIR` for a retention period of **6 months**.
+  5. **Reporting:**
+     - Reports status back to the PR checks.
+     - Generates and uploads HTML reports.
+- **Artifacts:** Test reports, logs, screenshots, accessible via PR details under **Artifacts**.
+- **Notifications:** Status updates posted on the PR.
+- **Manual Retriggering:**
+  - Tests can be manually retriggered using the `/retest e2e-tests` or `/test all` commands in the PR comments.
+
+### GitHub PRs Diagram
+
+![GitHub PR Testing Diagram](./resources/github_diagram.svg)
+
+## Nightly Tests
+
+Nightly tests are run to ensure the stability and reliability of our codebase over time. These tests are executed on different clusters to cover various environments, including both **RBAC** and **non-RBAC**.
+
+- **AKS Nightly Tests:** Nightly tests for Azure Kubernetes Service (AKS) run on the `bsCluster`. We do not have AKS PR checks; the AKS environment is exclusively used for nightly runs.
+
+- **IBM Cloud Tests:** All nightly tests for the `main`, `1.3`, and `1.2` branches run against the `rhdh-pr-os` OpenShift Container Platform (OCP) cluster on IBM Cloud.
+
+### Additional Nightly Jobs for Main Branch
+
+The nightly job for the `main` branch also runs against:
+
+- **`rhdh-os-1`** (currently OCP 4.14).
+- **`rhdh-os-2`** (currently OCP 4.15).
+
+> **Note:** The output of the nightly runs, including test results and any relevant notifications, is posted on the Slack channel **`#rhdh-e2e-test-alerts`**.
+
+### CI Job Definitions
+
+#### Nightly Test Job
+
+- **Purpose:** Ensure ongoing stability and detect regressions in different environments.
+- **Trigger:** Scheduled to run every night.
+- **Environments:**
+  - **AKS Nightly Tests:** Runs on the `bsCluster`.
+  - **IBM Cloud Nightly Tests:** Runs on the `rhdh-pr-os`, `rhdh-os-1`, and `rhdh-os-2` clusters.
+- **Configurations:**
+  - Tests are executed on both **RBAC** and **non-RBAC** namespaces.
+- **Steps:**
+  1. **Triggering:** Nightly job is triggered on schedule.
+  2. **Environment Setup:** Uses the `openshift-ci-tests.sh` script for setting up the environment.
+     - **Cluster Selection:** Chooses the appropriate cluster based on the job name.
+     - **Resource Configuration:** Sets up namespaces and configures resources.
+     - **Deployment:** Deploys the Red Hat Developer Hub (RHDH) instance and necessary services.
```

Comment:
@nilgaar lets add what all nightly jobs we have and what rhdh image tag/version is deployed including the quay image repository url.

Reply:
in the nightly test section we stated:
`### Nightly Test Environments

- **AKS Nightly Tests:** Nightly tests for Azure Kubernetes Service (AKS) run on a dedicated cluster. We do not have AKS PR checks; the AKS environment is exclusively used for nightly runs.
- **IBM Cloud Tests:** All nightly tests for the `main`, `release-1.3`, and `1.2.x` branches run against OpenShift clusters on IBM Cloud.`

Is there anything additional that I'm missing?
I am afraid I don't know exactly where to find the image tag/version deployed and the quay image url. May you help me with this? Thank you!